### ISSUE TITLE
How to set custom loss function in StructuredDataClassifier?

### ISSUE THREAD

**Issue by mc-karsa-tech** [keras-team/autokeras#1608]
Hi,

we are doing classification prediction. Our dataset is imbalanced and contains much more zero labels than ones.
We are trying to solve this by our custom loss function which returns percent of errors in either true zeroes or true ones which is larger.
For example, if there is 5% error in true zeroes and 20% error in true ones, than loss function returns the larger error i.e. 20.

```
def custom_function_max_incorrect(y_true, y_pred):
    y_true = backend.round(backend.clip(y_true, 0, 1))
    y_pred = backend.round(backend.clip(y_pred, 0, 1))

    all_zeroes_count = backend.maximum(backend.sum(1.0 - y_true), 1) # must be more than 0
    incorrect_zeroes_count = backend.sum((1.0 - y_true) * y_pred)
    incorrect_zeroes_percents = 100.0 * incorrect_zeroes_count / all_zeroes_count

    all_ones_count = backend.maximum(backend.sum(y_true), 1) # must be more than 0
    incorrect_ones_count = backend.sum(y_true * (1.0 - y_pred))
    incorrect_ones_percents = 100.0 * incorrect_ones_count / all_ones_count

    incorrect_percents_max = backend.maximum(incorrect_ones_percents, incorrect_zeroes_percents)
    return incorrect_percents_max
```
    
The only problem is that `StructuredDataClassifier` seems to ignore our loss parameter and instead always uses a default one.
This is our code:

```
    clf = autokeras.StructuredDataClassifier(
        overwrite=True,
        max_trials=200,
        loss = custom_function_max_incorrect,
        objective = kerastuner.Objective("custom_function_max_incorrect", direction="min")
    )
    
    clf.fit(train_file_path, "CurrentHasOrder", epochs=100)
```
    
Interestingly, `StructuredDataRegressor` is correctly using loss parameter which we provide, but `StructuredDataClassifier` seems to ignore it.

We also tried to declare custom object but `StructuredDataClassifier` was still using the default loss function and ignored our custom loss function:

```
    with tf.keras.utils.custom_object_scope({'custom_function_max_incorrect': custom_function_max_incorrect}):

        clf = autokeras.StructuredDataClassifier(
            overwrite=True,
            max_trials=200,
            loss = custom_function_max_incorrect,
            objective = kerastuner.Objective("custom_function_max_incorrect", direction="min")
        )
        
        clf.fit(train_file_path, "CurrentHasOrder", epochs=100)
```

How to use custom loss function in `StructuredDataClassifier`, please?

**Comment by haifeng-jin** [comment]
We don't have this feature yet. Welcome to discuss on what the API should look like.
